kornia/feature/siftdesc.py

```diff
@@ -77,12 +77,25 @@ class SIFTDescriptor(nn.Module):
     """
 
     def __repr__(self) -> str:
-        return self.__class__.__name__ +\
-            '(' + 'num_ang_bins=' + str(self.num_ang_bins) +\
-            ', ' + 'num_spatial_bins=' + str(self.num_spatial_bins) +\
-            ', ' + 'patch_size=' + str(self.patch_size) +\
-            ', ' + 'rootsift=' + str(self.rootsift) +\
-            ', ' + 'clipval=' + str(self.clipval) + ')'
+        return (
```

Comment:
@asottile is there any way to not destroy that much this ones ? @carmocca was suggesting to use f-strings, which possibly means revisiting big part of the library

Reply:
yeah I would probably move those to f-strings -- though you might be able to get away with a little helper that's something like this:

```python
return ''.join((
    f'{type(self).__name__}(',
    ', '.join(f'{name}={getattr(self, name)}' for name in sorted(self.__dict__) if not name.startswith('_')),
    ')'
))
```

(warning: written in github ui, might need some tweaking -- but the basic idea is to pull the attributes off of the object instead of having to hardcode all of them)

Reply:
cool , thanks. We'll try to address that  /cc @shijianjian

Reply:
https://github.com/kornia/kornia/issues/1053